### DISCUSSION TITLE
Is it possible to load and build only a sub-module in a multi module project

### DISCUSSION THREAD

**Discussion by yadavan88** [sbt/sbt#7356] [Q&A]
HI,
this might be a very silly question. 
I want to know if there is a way to load and build only a single sub-module directly, without loading the entire project. This is a bit different structure to the normal projects I am used to.
There is a tutorial project that already has 30+ sub modules. All these sub-modules are defined in the main build.sbt. If someone want to run a test/main class in ide, one needs to open the entire project in ide and wait for it to load and download entire structure. To do this, I tried having each module dependencies inside each module's build.sbt(it didn't exist before, but I added). 
It works, however, there is a lot of dependency and version duplication since they need to be given inside the sub-module's build,sbt. Is there any way to avoid this duplcation, but still load only a single module with build?

Note: We should be able to load and build the entire module (from the root level) as well. There are some aliases defined to build and run tests across all the 30+ sub modules in CI env

Thanks in advance

**Comment by gontard** [comment]
This discussion might be of interest for you:
https://github.com/sbt/sbt/discussions/7214
